core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -1922,6 +1921,18 @@ public SqlNode field(int ordinal, boolean useAlias) {
           needNew && !aliases.containsKey(neededAlias) ? newAliases : aliases);
     }
 
+    private boolean isAliasRefNeeded(boolean keepColumnAlias, RelNode rel) {
+      if (expectedClauses.contains(Clause.HAVING)
+          && dialect.getConformance().isHavingAlias() || keepColumnAlias) {
+        return true;
+      }
+      if (expectedClauses.contains(Clause.QUALIFY)
+          && rel.getInput(0) instanceof Aggregate) {
+        return true;
```

Comment:
QUALIFY clause evaluation order change to after SELECT, then this should not be required.

Reply:
These logic is related for alias reference if we have any aggregate function inside QUALIFY clause we need to pass the alias reference and if we change the evaluation order also need to write this logic